pageserver/src/tenant/timeline.rs

```diff
@@ -1845,12 +1873,24 @@ impl Timeline {
                     Ok(s) => s,
                     Err(CalculateLogicalSizeError::Cancelled) => {
                         // Don't make noise, this is a common task.
-                        // In the unlikely case that there ihs another call to this function, we'll retry
+                        // In the unlikely case that there is another call to this function, we'll retry
                         // because initial_logical_size is still None.
                         info!("initial size calculation cancelled, likely timeline delete / tenant detach");
                         return Ok(());
                     }
-                    x @ Err(_) => x.context("Failed to calculate logical size")?,
+                    Err(CalculateLogicalSizeError::Other(err)) => {
+                        if let Some(PageReconstructError::Stopping) =
+                            err.root_cause().downcast_ref()
+                        {
+                            // This can happen at tenant detach, when the parent timeline switches
+                            // to Stopping state while we're still calculating the initial timeline
+                            // size for the child. Like in the CalculateLogicalSizeError::Cancelled,
+                            // don't make noise.
+                            info!("initial size calculation failed because the timeline or its ancestor is Stopping, likely because the tenant is being detached");
+                            return Ok(());
+                        }
```

Comment:
... or because we're restarting pageserver

Also, please print the error in the info! message using `{e:#}`.
If the comment above it is wrong about when this can happen, at least we'll have the error message.

Also, with my request to add a timeline_id field to the Stopping variant, printing the error will give us more breadcrumbs.

I think you can use the `if let Some(x @ PageReconstructError::Stopping(_))` syntax to get ahold of the error.

Reply:
Currently, the PageReconstructError::Stopping doesn't carry extra information, so printing it  doesn't really add anything. I did it anyway, so that if we add extra information to it in the future, it becomes useful and we don't forget to add the printing at that point.